### DISCUSSION TITLE
Command Line support

### DISCUSSION THREAD

**Discussion by Midnex** [rustdesk/rustdesk#3980] [General]
Would like to see command line support so I can add RustDesk to env variables, and push an id as an argument along with ability to push a password if needed.

Of course the command line interface would need to support other things as well, but this support would be a nice addition.

This feature could also be used to drop shortcuts on to the users desktop, that calls RustDesk and connects to a saved id in the shortcut file. Similar to how AnyDesk does it.

**Comment by rustdesk** [comment]
> drop shortcuts

1.1.6

**Comment by rustdesk** [comment]
https://support.anydesk.com/Command_Line_Interface

**Comment by Midnex** [comment]
Glad to see this reopened.

**Comment by rustdesk** [comment]
Because someone mentioned this on discord, and I have been investigating on AnyDesk/TeamViewer/Parsec API lately.

**Comment by radiobut** [comment]
It would be very nice to be able to start a connection from the command line. For example, to integrate with other remote connection management tools.

**Comment by bilogic** [comment]
Teamviewer https://community.teamviewer.com/English/kb/articles/34447-command-line-parameters

I normally use `teamviewer.exe -i <id>`, and it prompts for password, very useful if there are many machines to manage.

**Comment by rustdesk** [comment]
`--connect <id>`

**Comment by bilogic** [comment]
> `--connect <id>`

wow thanks! is it documented anywhere?

**Comment by Shahin-rmz** [comment]
Hello,
A power user (not a dev) here.
First:
Rustdesk is WAY better than Anydesk or Teamviewer.
I have installed it on my parent's Raspberry pi.
and can control the Raspberry remotely and play Movie for them.
Thanks <3

### Now a question:
_________________________________________________________________
Can I connect via SSH and generate a password, (or see the password) and then connect via client to the Raustdesk?
_____________________________________________________________________________

Thanks

**Comment by bilogic** [comment]
> Can I connect via SSH and generate a password, (or see the password) and then connect via client to the Raustdesk?

Are you suggesting this is more secured than having a static complex password on rustdesk?

Frankly, I think it takes the same flow and effort to steal your static complex password vs your SSH password/key.

**Comment by Shahin-rmz** [comment]
> Frankly, I think it takes the same flow and effort to steal your static complex password vs your SSH password/key.


You are right.
Ultimately I like to:
-  have a dynamic password for **every new connection every time** but a static password for trusted connections.
Or other way around,
-  trusted connections make a pair of password each time.

**Comment by bilogic** [comment]
Haha, we are going in circles: how do you determine if a connection is a trusted one?

Teamviewer does it by having 2 passwords, a dynamic one and static one, either one will allow you to login.

The closest I can think of is to have the dynamic password regenerate after each disconnection, but the static remains and is available to all connections, just make sure to keep it complex.

**Comment by Shahin-rmz** [comment]
> Haha, we are going in circles: how do you determine if a connection is a trusted one?

You are right. :) I am thinking that if both parties *check the trusted box*, then we the connections are *trusted*.

> The closest I can think of is to have the dynamic password regenerate after each disconnection, but the static remains and is available to all connections, just make sure to keep it complex.

This is one good approach, the other one is to have static password just for some connections and **not global**.

**Comment by paulbdavis** [comment]
Jumping in this issue to say it would be good to be able to get the current ID and password via the command line. 

The use case is having a machine that's out in the field where there is not someone present to read off the ID and password or interact with the UI component (the UI would not even be launched). It would be good to dump these out on the command line when connected to said machine via SSH

like:

```
$ rustdesk --info

ID: 123 123 123
Password: abcdef
```

**Comment by paspo** [comment]
you can read ID and password from the config files:

- C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk.toml (windows)
- /root/.config/rustdesk/RustDesk.toml (linux)
- /Library/Preferences/com.carriez.RustDesk/RustDesk.toml (OSX)

(password can be salted)

**Reply by 593172440** [reply]
1.2.0 ID encryption。How else can I obtain it？

**Reply by mooleshacat** [reply]
No you cannot. 

```
enc_id = '00C4zmeQKHtMzP+cxTTLMQaLSObEIVVWl8'
password = ''
```

As you can see, the ID is encrypted and the password is non-existant.

**Comment by rustdesk** [comment]
will be encrypted in next release

**Comment by bilogic** [comment]
> Jumping in this issue to say it would be good to be able to get the current ID and password via the command line.

I think it is generally not safe to expose the ID and password via an "API" approach. Anything else can also call the "API" and gain access. Getting a dynamic password via "API" using SSH is equivalent to setting a static password (SSH being the static in this case).

It is probably best to just let Rust Desk do its best to protect the static password instead of making it leak credentials via an "API".

**Comment by paulbdavis** [comment]
For the password, sure, I suppose that makes sense, but getting the ID and being able to *set* the password would be enough really.

Having it disabled by default would obviously be best as well, but there should be a way to get/set the credentials for a remote machine without anyone actually being present.

**Comment by bilogic** [comment]
> Having it disabled by default would obviously be best as well, but there should be a way to get/set the credentials for a remote machine without anyone actually being present.

In that case, I think it is better to state the problem ("How to perform unattended remote installation") instead of suggesting the solution.

In one of the VPS that I use, they ask for my SSH public key when provisioning the server. i.e. unattended installs must have a static password, but please don't quote me on that, authentication/credential modifications must always be rigorously thought out.

**Comment by jcelerier** [comment]
>  In that case, I think it is better to state the problem ("How to perform unattended remote installation") instead of suggesting the solution.

what's the solution to this problem? how can I get a remote desktop remotely from a machine which has nothing installed and to which I have a ssh access ?

**Comment by paulbdavis** [comment]
Sure, for the password that is fine, I can just set it manually, but what about getting the ID? Will that also be encrypted on the next release?

**Comment by rustdesk** [comment]
Yes， enced in next release.

**Comment by ssteeltm** [comment]
would be nice if I could connect via CMD with:
`` rustdesk.exe --connect <ID> --password <password> ``

**Comment by bilogic** [comment]
`--connect <id>`, use rustdesk to remember the password

**Comment by Onyx47** [comment]
Lack of an option to retrieve the ID programmatically, possibly from a file or as a command-line option that can only be read/ran as root/elevated user is what's currently stopping me from replacing AnyDesk with RustDesk for my use case.

Having the ID recorded somewhere where it's only accessible to root/admin does not strike me as a huge security risk (AnyDesk does this as well, as an aside), and enables easier automation for cases where you already have some kind of elevated access available to you and you just need that extra step of being able to access the GUI.

Also, just to point out, when starting RustDesk from terminal on Linux, it still spits out the ID to `stdout`, so it's technically still possible to grab the ID programatically, but this is not something I'd rely on, plus it's inconsistent - it's encrypted in a file but it's readable by a regular user if they can just execute a command as an unprivileged user.

**Comment by mikelwellsmore** [comment]
Agreed - we need a way to programmatically retrieve or set the ID (as it looks like the password can be set via `rustdesk.exe --password`)

**Comment by kapytanhook** [comment]
we need to be able to set or read the ID or there is no point to this software, going to have to make my own build just to do that now.

**Comment by Onyx47** [comment]
For anyone needing this functionality and is willing to build their own release, this is my current solution until there is a more definite decision on this: https://github.com/Onyx47/rustdesk/commit/ed9e92fc2e46fb3272cb4d694802ac8915a45ba7

It enables you to run `rustdesk --id` as admin/root and simply prints out the ID and quits. Certainly not sophisticated but it should do for scripting purposes. Only tested on Linux but I'd assume Windows should be fine as well, it's using the same logic as `--password` argument.

**Comment by knro** [comment]
It would be nice to be able to set/get ID as root. Any ideas if this will ever be merged in Rustdesk or we'd have to rollout our own releases?

**Comment by merlink01** [comment]
For autologin the way anydesk does it is nice:
Load the password from stdin.

So the command is:
`rustdesk.exe --connect <ID> --password`
and after this:
send the password via stdin to the process followed by a \n.

On this way the password is not visible on the process list.

**Reply by PierreCsn** [reply]
would be great to add rustdesk support into other products

**Reply by PrieserMax** [reply]
Sadly I am a noob at scripting.
I have a working script like this (batch-file):

C:
cd C:\Program Files\RustDesk
start "" rustdesk.exe --connect ID --password "PASSWORD"

How would I pass through the password as you elaborated?
Greetings
Max

**Comment by knro** [comment]
> Jumping in this issue to say it would be good to be able to get the current ID and password via the command line.
> 
> The use case is having a machine that's out in the field where there is not someone present to read off the ID and password or interact with the UI component (the UI would not even be launched). It would be good to dump these out on the command line when connected to said machine via SSH
> 
> like:
> 
> ```
> $ rustdesk --info
> 
> ID: 123 123 123
> Password: abcdef
> ```

This is exactly our use case. We need to a use to programatically retrieve the ID. We can set password it's sad to see no way to get ID for field deployments. It's OK if this action is only limited to root, that's not the issue. @rustdesk When can we expect this important feature to be implemented? https://github.com/Onyx47/rustdesk/commit/ed9e92fc2e46fb3272cb4d694802ac8915a45ba7 seems like a good solution.

**Reply by ksingh7** [reply]
@knro  did this worked for you ?

**Comment by shanjunmei** [comment]
Highly recommend this, I can only precompile one client per client right now, built in id and password ,is there any body can share a document ,how to pre-config it

**Comment by gezusk** [comment]
I saw the option to rename the Windows installer file to include the relay server and public key. Is it possible to provide those via cmd line instead of creating a really....really long filename?

**Reply by mooleshacat** [reply]
I think that method was jenk
They should have allowed command line args.

**Comment by christovic** [comment]
Just chiming in here to say that this is really, really useful in managed deployments with MDM.

**Comment by SergeyMy** [comment]
so it's all done!
remotely via ssh:
1. installation - yes
2. rendezvous_server setting - yes
3. relay-server setting - yes
4. api-server setting - yes
5. setting KEY_PUB - yes
6. get ID - yes (--get-id)
7. set a permanent password - yes (--password)

**Reply by christovic** [reply]
So you are right! Didn't see this anywhere in the release notes.

Thanks

**Reply by gezusk** [reply]
On Windows, those don't appear to return anything. I tried --get-ID and --version, and nothing comes up. I tried -get-id, and that opens the RustDesk window...so I assume that's wrong.

**Reply by dinger1986** [reply]
they do, use cmd not powershell :) `"c:\Program Files\RustDesk\RustDesk.exe" --get-id`

I have no idea why it doesnt work in powershell but it works well in cmd

**Reply by SergeyMy** [reply]
in Windows everything works a little strange, especially everything related to privileges
**--get-id, --password** require real superuser (administrator++) privileges
:)

**Reply by gezusk** [reply]
I was using command line.

I'm an administrator on my system, but I also tried 'runas Administrator'. I'm not sure how to elevate even higher :)

**Reply by dinger1986** [reply]
What version of rustdesk?

**Reply by gezusk** [reply]
1.2 Stable.

**Reply by 0x100001** [reply]
Same for me. Tried with Powershell and CMD. Neither is working. The code is existing tho.
https://github.com/rustdesk/rustdesk/blob/6c8c3b4a7a3f3eb3570a6b15641447b0744673aa/src/core_main.rs#L225

**Reply by SergeyMy** [reply]
```
} else if args[0] == "--password" {
            if args.len() == 2 {
                if crate::platform::is_root() {
                    crate::ipc::set_permanent_password(args[1].to_owned()).unwrap();
                } else {
                    println!("Administrative privileges required!");
                }
```
what does he write?
PS 
` if args.len() == 2 {`
i.e. you need:
`./rustdesk --password PSW999888dvaraza PSW999888dvaraza`
:):):):):):):):)

**Reply by 0x100001** [reply]
@rustdesk I think you might have a logic problem there.

Code: https://github.com/rustdesk/rustdesk/blob/6c8c3b4a7a3f3eb3570a6b15641447b0744673aa/src/core_main.rs#L225
Thats the code to check whetever the parameter is called or not:

else if args[0] == "--get-id" {
    if crate::platform::is_root() {
        println!("{}", crate::ipc::get_id());
    } else {
        println!("Permission denied!");
    }

You call:
if crate::platform::is_root()

but the function is_root() does the following:
pub fn is_root() -> bool {
    crate::username() == "root"
}

That means if your user is not called "root", it will not give back the id, the other parameters wont work too. That cant work properly on windows. You might need to correct me if I am wrong, I just checked that code parts.

**Reply by SergeyMy** [reply]
ЗЫ хахаха разработчики шутники....
обычный пользователь в windows! никогда так не сможет!

According to the source code in Windows, this works as the **system user!** ordinary mortals can't do it!
![изображение](https://github.com/rustdesk/rustdesk/assets/131688106/cecc2a43-3df4-4287-80a4-c57c8c233c56)

**src/platform/windows.rs**
```
pub fn is_root() -> bool {
    // https://stackoverflow.com/questions/4023586/correct-way-to-find-out-if-a-service-is-running-as-the-system-user
    unsafe { is_local_system() == TRUE }
}

```

**Reply by 0x100001** [reply]
@SergeyMy thanks for the reply! I was pretty stupid and clicked on the linux source... I also didnt try to implement it into my windows service yet, cause normal admin user tests failed. I think @dinger1986 executed it through a monitoring software (RMM), causing the cmd to have local system priviliges. In the evening I will test it with my windows service and give feedback.

**Reply by gezusk** [reply]
Sooo...I guess the command isn't meant to ever be used? Since no one should be running as system user, or root.

**Reply by gezusk** [reply]
I don't even get an error message about permissions though. Just nothing...

**Reply by 0x100001** [reply]
@gezusk you need a (normal usecase) windows service running with local system user privlieges to execute the parameters. If you are looking for a solution to read id & password for central management of rustdesk, feel free to contact me, or contact @dinger1986 for information about the pro version of rustdesk.

**Reply by SergeyMy** [reply]
these commands work fine on linux when deploying rustdesk remotely via ssh

**Reply by gezusk** [reply]
@0x100001 I'm interested in pro...but can't find much information about it, or a demo.

**Reply by SergeyMy** [reply]
if you need to install on more than a couple of systems .... it's strange that there is no smart programmer even in the web ... in the PRO version you will get at least a web api, where all clients themselves will report their IDs
PS
you don't even need a programmer to find out the ID! just look at sqlite db :) on yours!!! server hbbs
`sqlite3 -column db_v2.sqlite3 "select id,created_at,user,note,info,status,uuid,hex(guid) from peer;"`

**Reply by 0x100001** [reply]
@gezusk contact: sales@rustdesk.com
They will help you. 

@SergeyMy I do that. There is a free version of my monitoring software just for that. Feel free to contact me.

**Reply by AlexandreBonneau** [reply]
Using ssh, I was able to install the flatpak version of rustdesk on a Debian box.
I can run `--get-id` to retrieve the id, however when I try to run `flatpak run com.rustdesk.RustDesk --password foobar`, I get the following error message:
`Installation and administrative privileges required`.
I tried running the same command with root, to no avail.
Any ideas how to fix that?

**Reply by shahg2** [reply]
Hi alexandre, did you find the solution to this. iam facing this issue in Windows. Normal rustdesk.exe password can be set but if i customized the rustdesk client and run cli --password i get this message. whereas with rustdesk.exe -- password i can set password and it will show done. but with custom client it will show "Installation and administrative privileges required!"

**Comment by SergeyMy** [comment]
> @SergeyMy I do that. There is a free version of my monitoring software just for that. Feel free to contact me.

you need? help with developing and testing web api pro lite? ( ;) smile)
```
<?php
......
//********************************************* MAIN POST ********************
if($islog) foreach ($_SERVER as $key => $val)    {    fwrite($file, $key . ' => ' . $val . "\n");    }
foreach ($_REQUEST as $key => $val)
 {
    if($islog) fwrite($file, $key . ' => ' . $val . "\n");

if (stristr($key, 'api/heartbeat'))   {
    header('HTTP/1.1 200 Ok');
    header('Content-Type: application/json');
    $arr = array('modified_at' => 12);
    echo json_encode($arr);
    }
}
?>
```

**Reply by 0x100001** [reply]
I think I might didn't explain it correctly. I coded a RustDesk integration for my monitoring software, including monitoring the RustDesk service, enforcing a custom relay server, reading id & password, generating a password and stuff like that. I sadly dont have the time capacities to work on other projects than mine. :/

**Comment by chris-31337** [comment]
Hi all, do you know if there has been progress on password-passing via command-line during `--connect`, preferably via Stdin as suggested by merlink01 above and as implemented in Anydesk? 

While I am currently following bilogic's suggestion above to store the password in Rustdesk, I would like to store connection files (e.g. batch files to execute `rustdesk.exe --connect ID` plus password) for specific hosts in an encrypted volume for added security (without relying on full-device encryption to encrypt ALL passwords stored by Rustdesk under %AppData%\RustDesk\config\peers in Windows).

**Reply by dinger1986** [reply]
https://rustdesk.com/docs/en/client/#command-line-parameters

You can do all you have asked about

**Reply by chris-31337** [reply]
Thank you for the swift response! However, the link you provided does not seem to explain a way to pass the password to rustdesk via command-line. Indeed, while `--password` intuitively sounds like what I have asked for, the doc contrarily says:
> `--password` can be used to set a permanent password.

Which does not sound like what I need. Did I miss something in your link? Thank you!

**Reply by dinger1986** [reply]
The link then links to the code where all the command options are

**Reply by chris-31337** [reply]
Thank you! I went through that code already but apparently missed the right command-line to pass a password. Could you point me in the right direction?

**Reply by dinger1986** [reply]
I'd have to read through the code to remind myself, I'm sure it's something like `rustdesk.exe --connect id password=password` if you get it working feel free to add to the docs for the next person

**Reply by chris-31337** [reply]
Thank you! In my ignorance, I did not expect `password` to have a dual function in addition to what is written in the docs. As you suggested, `RustDesk.exe --connect ID --password PASSWORD` works.

However, I am still curious if there is a way to more securely pass the password using Stdin as already suggested by merlink01, since the password supplied via `--password` can be read from memory while the connection is open. `cmd /c echo PASSWORD | RustDesk.exe --connect ID` or `cmd /c echo PASSWORD | RustDesk.exe --connect ID --password` (which is how I pass the password via Stdin to Anydesk) did not work in my tests.

**Reply by ivoryblade** [reply]
Hello! Let me join your discussion. Tell me please is there any way to pass arguments on Mac/linux clients? And it would be most wanted if one can use some universal way to start connection from external remote management software win/Mac/linux

**Comment by plittlefield** [comment]
Hi Folks,

This would be great.

I tried rustdesk --help and it just fires up the GUI in X again.

Is there a document for all the command line options on Linux please?

Paully

**Reply by dilawar** [reply]
I am also baffled about this behaviour? `sudo rustdesk --get-id` prints the id of the desk but without `sudo`, it tries to launch the app!!

**Reply by thiswillbeyourgithub** [reply]
FYI I made a FR to ask for a --help argument at #9484

**Comment by thiswillbeyourgithub** [comment]
For anyone still wondering what command line arguments are currently available, take a look at [the file src/core_main.rs](https://github.com/rustdesk/rustdesk/blob/master/src/core_main.rs) and do ctrl+f for '--'.

I'm not sure there are other arguments than those.


Edit: in  #9484 I made a request to add a --help argument

**Reply by gitterspec** [reply]
> For anyone still wondering what command line arguments are currently available, take a look at [the file src/core_main.rs](https://github.com/rustdesk/rustdesk/blob/master/src/core_main.rs) and do ctrl+f for '--'.
> 
> I'm not sure there are other arguments than those.
> 
> Edit: in #9484 I made a request to add a --help argument

Thanks for this. However, I can't seem to find the values that --option can take. The code seems to call get_options but looking at that function doesn't seem to give much insight. Anyone have some idea?

**Comment by lazee486** [comment]
go to the network tab in rustdesk after youve added your server and key, click the export button on top right...gives you a encoded string

then run rustdesk.exe --config =0nI9Ajeh9GWRdlSJBFc0F1VpZDTCxUYjlEeItkWxZWZnxUWBd3Srp0Z4gFWNxkI6ISsdfhshsdhsdhIiI6ISehxWZyJCLiEmeu82YuQXa5xGcdhdhdhjsdfhciojI0N3boJye

thats a garbage key just as an example...  https://rustdesk.com/docs/en/self-host/client-deployment/   has the scripts - use that same string you exported in the config = " " field to preconfigure for your deploys.

**Reply by lazee486** [reply]
rustdesk.exe --config =0nI9Ajeh9GWRdlSJBFc0F1VpZDTCxUYjlEeItkWxZWZnxUWBd3Srp0Z4gFWNxkI6ISsdfhshsdhsdhIi

shortened it so people can see its meant to be a single line command 

other useful
rustdesk.exe --silent-install
rustdesk.exe --install-service
rustdesk.exe --get-id            (returns id)
rustdesk.exe --password myshittypassword                       (sets password)

**Comment by rustdesk** [comment]
```
rustdesk.exe --connect <id> --password <password>
rustdesk.exe --config <config-string>
rustdesk.exe --silent-install
rustdesk.exe --install-service
# get id
rustdesk.exe --get-id
# set password
rustdesk.exe --password myshittypassword
# set id
rustdesk.exe --set-id <id>
```

> please run `--config`, `--password`, `--set-id` with administrative privilege, and make sure RustDesk is installed rather than run as portable version.

By default rustdesk.exe in command line on Windows has no output, you have to run it like below.
- cmd.exe: `rustdesk.exe <args> | more`, e.g. `rustdesk.exe --get-id | more`
- PowerShell: `rustdesk.exe <args> | Out-String`, e.g. `rustdesk.exe --get-id | Out-String`

# Assign Device Users/Groups/Strategies to Devices for self-hosting Pro

https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/console/#assign-device-usersgroupsstrategies-to-devices

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/blob/faf97c770cc5a3a2c65a9b9734c37f57ea493e44/src/core_main.rs#L174

https://github.com/rustdesk/rustdesk/blob/faf97c770cc5a3a2c65a9b9734c37f57ea493e44/src/core_main.rs#L573

**Reply by gxsalles** [reply]
Thanks for the tip, however ,using "CMD" it was necessary to include .\ at the beginning of the command

.\rustdesk.exe --password myshittypassword

**Reply by wadson-dev** [reply]
> ```
> rustdesk.exe --connect <id> --password <password>
> rustdesk.exe --config <config-string>
> rustdesk.exe --silent-install
> rustdesk.exe --install-service
> # get id
> rustdesk.exe --get-id
> # set password
> rustdesk.exe --password myshittypassword
> # set id
> rustdesk.exe --set-id <id>
> ```
> 
> > please run `--config`, `--password`, `--set-id` with administrative privilege, and make sure RustDesk is installed rather than run as portable version.
> 
> By default rustdesk.exe in command line on Windows has no output, you have to run it like below.
> 
> * cmd.exe: `rustdesk.exe <args> | more`, e.g. `rustdesk.exe --get-id | more`
> * PowerShell: `rustdesk.exe <args> | Out-String`, e.g. `rustdesk.exe --get-id | Out-String`
> 
> # Assign Device Users/Groups/Strategies to Devices for self-hosting Pro
> https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/console/#assign-device-usersgroupsstrategies-to-devices

rustdesk.exe --password myshittypassword

Ran this command as elevated user and had no change in password, still using one-time password. Was this expected? 
In resume, no change at all

![image](https://github.com/user-attachments/assets/1ed933ca-099c-4bcd-9454-6a24f92df1fc)
![image](https://github.com/user-attachments/assets/5a59616c-214d-49f2-90c5-0509b52bc970)

**Comment by 621625** [comment]
I second this. Rustdesk should have a CLI. I'm on Linux. I want it and I expect it. And I love rustdesk.

**Comment by mitenpetal** [comment]
Can someone please help me to setup a client on Linux machine ? 
what I want is config added , add to addressbok , add to device group and , user group and last password add Is it possible ? 
Please help me with terminal command , I am using rocky linux 

thank you

**Reply by mitenpetal** [reply]
https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/console/#assign-device-usersgroupsstrategiesdevicegroup-to-devices
Thank you this way it is working. 

It has worked but not entirely. Which group has space in between names for that it is not working, Is there Any solution for that?

**Comment by austinbutler** [comment]
Guessing there's no way to accept the share prompt via SSH/command line?

**Comment by shahg2** [comment]
I am using a custom-built RustDesk client named customclient.exe.The client is successfully installed as a service and the service is actively Running.I am executing the command with full Administrative privileges to set permananet password. 
However, when I run the standard command, I receive an error, while the original RustDesk.exe works fine: Custom customclient.exe Failure: .\customclient.exe --password XXX | more $\rightarrow$ Installation and administrative privileges required!
Since the service is confirmed running, the issue seems to be an internal check failure within the custom build.What specific component in the RustDesk source code might be missing, or what configuration key might be different in the custom build that prevents the CLI command from recognizing the installed service and updating the configuration file?
Specifically, is there a hardcoded internal check for the executable name or a service name/path mismatch that causes the --password command to fail its service-status check?
